packages/@sanity/migrate/src/fetch-utils/fetchStream.ts

```diff
@@ -18,9 +18,17 @@ export async function assert2xx(res: Response): Promise<void> {
   if (res.status < 200 || res.status > 299) {
     const jsonResponse = await res.json().catch(() => null)
 
-    const message = jsonResponse?.error
-      ? `${jsonResponse.error}: ${jsonResponse.message}`
-      : `HTTP Error ${res.status}: ${res.statusText}`
+    let message: string
+
+    if (jsonResponse?.error) {
+      if (jsonResponse?.error?.description) {
+        message = `${jsonResponse?.error?.type || res.status}: ${jsonResponse.error.description}`
+      } else {
+        message = `${jsonResponse.error}: ${jsonResponse.message}`
+      }
+    } else {
+      message = `HTTP Error ${res.status}: ${res.statusText}`
```

Comment:
Could be nice to refactor to `res.text()` then put the text here in case the response isn't json

something like

```js
const textResponse = await res.text()
let jsonResponse
try {
  jsonResponse = JSON.parse(textResponse)
} catch {
  jsonResponse = null
}

// use textResponse etc...
```

Reply:
hmm in terms of lines of code that would be the same right? I don't see a reason API responds with text instead of JSON 🤔

Reply:
yeah probably over-thinking it